cime/src/drivers/mct/main/prep_glc_mod.F90

```diff
@@ -607,6 +777,156 @@ end subroutine prep_glc_map_one_state_field_lnd2glc
 
   !================================================================================================
 
+  subroutine prep_glc_calculate_subshelf_boundary_fluxes
+
+    !---------------------------------------------------------------
+    ! Description
+    ! On the ice sheet grid, calculate shelf boundary fluxes
+
+    use shr_const_mod , only: SHR_CONST_KAPPA_LAND_ICE
+
+    ! Local Variables
+
+    integer :: gsize, n, egi
+    type(mct_aVect), pointer :: o2x_ox ! Ocn export, ocn grid, cpl pes
+    type(mct_aVect), pointer :: x2g_gx ! Glc import, glc grid, cpl pes
+    type(mct_aVect), pointer :: g2x_gx ! Glc import, glc grid, cpl pes
+
+    integer :: index_x2g_So_blt
+    integer :: index_x2g_So_bls
+    integer :: index_x2g_So_htv
+    integer :: index_x2g_So_stv
+    integer :: index_x2g_So_rhoeff
+    integer :: index_g2x_Sg_tbot
+    integer :: index_g2x_Sg_dztbot
+    integer :: index_g2x_Sg_lithop
+    integer :: index_g2x_Sg_icemask_floating
+
+    integer :: index_g2x_Sg_blis
+    integer :: index_g2x_Sg_blit
+    integer :: index_g2x_Fogx_qiceho
+    integer :: index_g2x_Fogx_qicelo
+    integer :: index_x2g_Fogx_qiceli
+    integer :: index_x2g_Fogx_qicehi
+
+    character(*), parameter :: subname = '(prep_glc_calculate_subshelf_boundary_fluxes)'
+    !---------------------------------------------------------------
+
+    if (.not.(glc_present)) return
+    !write(logunit,*) 'Starting prep_glc_calculate_subshelf_boundary_fluxes'
+
+    do egi = 1,num_inst_glc
+
+    o2x_ox => component_get_c2x_cx(ocn(egi))
+    g2x_gx => component_get_c2x_cx(glc(egi))
+    x2g_gx => component_get_x2c_cx(glc(egi))
+
+    !Remap relevant ocean variables to ice sheet grid.
+    !Done here instead of in glc-frequency mapping so it happens within ocean coupling interval.
+    ! Also could map o2x_ox->o2x_gx(1) but using x2g_gx as destination allows us to see
+    ! these fields on the GLC grid of the coupler history file, which helps with debugging.
+    call seq_map_map(mapper_So2g, o2x_ox, x2g_gx, &
+    fldlist='So_blt:So_bls:So_htv:So_stv:So_rhoeff',norm=.true.)
+
+    ! inputs to melt flux calculation
+    index_x2g_So_blt =    mct_avect_indexra(x2g_gx,'So_blt',perrwith='quiet')
+    index_x2g_So_bls =    mct_avect_indexra(x2g_gx,'So_bls',perrwith='quiet')
+    index_x2g_So_htv =    mct_avect_indexra(x2g_gx,'So_htv',perrwith='quiet')
+    index_x2g_So_stv =    mct_avect_indexra(x2g_gx,'So_stv',perrwith='quiet')
+    !index_x2g_So_rhoeff = mct_avect_indexra(x2g_gx,'So_rhoeff',perrwith='quiet')
```

Comment:
rhoeff is remapped above but then commented out here. Please clean this up.

Reply:
Fixed in 733a46925b5580e5888bdfa417add9916e7cf2df